torch/_dynamo/skipfiles.py

```diff
@@ -145,78 +157,109 @@ def _module_dir(m: types.ModuleType):
     return _strip_init_py(m.__file__)
 
 
-# TODO(ybliang): Change to user *.__file__ rather than hard code string for this list.
-# Force inline functions in these files, even the files is in *_SKIPLIST.
-FILENAME_INLINELIST = {
-    torch.nn.Sequential.__init__.__code__.co_filename,
-    torch.set_rng_state.__code__.co_filename,
-    torch._inductor.test_operators.__file__,
-    torch.utils._content_store.__file__,
-    external_utils.__file__,
-    comptime.__file__,
-    polyfill.__file__,
-    torch.optim._functional.__file__,
-    torch.utils._foreach_utils.__file__,
-    _module_dir(torch) + "ao/quantization/pt2e/qat_utils.py",
-    _module_dir(torch) + "ao/quantization/quantizer/xnnpack_quantizer.py",
-    _module_dir(torch) + "ao/quantization/pt2e/representation/rewrite.py",
-    _module_dir(torch) + "ao/quantization/pt2e/utils.py",
-    _module_dir(torch) + "ao/quantization/pt2e/eval_utils.py",
-    _module_dir(torch) + "_dynamo/_trace_wrapped_higher_order_op.py",
-    _module_dir(torch) + "_export/constraints.py",
-    _module_dir(torch) + "_higher_order_ops/cond.py",
-    _module_dir(torch) + "_functorch/apis.py",
-    _module_dir(torch) + "_functorch/deprecated.py",
-    _module_dir(torch) + "distributed/tensor/parallel/_utils.py",
-    _module_dir(torch) + "distributed/tensor/parallel/style.py",
-    _module_dir(torch) + "distributed/tensor/parallel/_data_parallel_utils.py",
-    _module_dir(torch) + "distributed/_tensor/api.py",
-    _module_dir(torch) + "distributed/_tensor/device_mesh.py",
+# TODO: Add a decoractor for easily adding functions to FUNC_INLINELIST
+# after resolving all circular import issues.
+FUNC_INLINELIST = set()
+
+
+# Force inline functions in these files or directories, even they are in *_SKIPLIST.
+# We are using python module name instead of file or directory object to avoid circular dependency.
+# Please keep this sorted alphabetically.
+FILE_INLINELIST = {
```

Comment:
It feels like it would be better to name this `SUBMOD_INLINELIST_***` where *** denotes whatever the difference here is. Also, it would be nice to know what exactly the blocker to merging them is.

Reply:
Yea, we should merge ```FILE_INLINELIST``` and ```SUBMODULE_INLINELIST```. The major blocker is there are two call sites of ```skipfiles.check_verbose``` are not consistent right now and we want preserve semantics in this PR. 

Call site 1:
https://github.com/pytorch/pytorch/blob/33403336fa14d56aab203ebecfa51ca04ed16cc0/torch/_dynamo/eval_frame.py#L522 

Call site 2:
https://github.com/pytorch/pytorch/blob/33403336fa14d56aab203ebecfa51ca04ed16cc0/torch/_dynamo/symbolic_convert.py#L2250

The difference is ```extrac_check```, where we check the ```SUBMODULE_INLINELIST```. I hit some latent bugs if making them consistent when prototyping, and fixed most of them already. I just want to leave this behavior change in a separate PR in case there is anything I didn't anticipate.

Reply:
I agree the better name you suggested, but I don't plan to change it now, since the PR of merging these two lists will be the next one.